Automattic/wp-calypso
# Account recovery: connect the redux state to <ResetPasswordForm>

## Summary
This PR connects the state and the actions to `<ResetPasswordForm>` component to make it functional.

## Test Plan
1. Open an incognito window, and visit http://calypso.localhost:3000/account-recovery/reset-password . You should see an empty card titled as "Reset Your Password".
1. Open the developer console, and run: `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST, userData: { user: 'southptestcall2' } } )`. The `user` parameter can be arbitrary, as long as it really exists. Now the form should be filled in like this:<br/> ![image](https://cloud.githubusercontent.com/assets/1842898/24148736/00d2baaa-0e7b-11e7-97da-17ff5b9818af.png)
1. Run `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA, userData: { user: 'southptestcall2' } } )` to fill the underlying redux state.
1. Choose one among the options, and hit "Continue". Note that nothing seems to happen on the view, but if you open the Network tab, you should see a POST request hit `/account-recovery/request-reset` endpoint with successful response.
1. Run `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_REQUEST_ERROR, error: {} } );` and you should see the error message like this: <br/>
![image](https://cloud.githubusercontent.com/assets/1842898/24149137/75d0cdaa-0e7c-11e7-8704-a065f8d7f07b.png)
